guides/bulk.md

```diff
@@ -0,0 +1,209 @@
+  await client.Indices.CreateAsync(books);
+}
+```
+
+## Bulk API
+
+The `bulk` API action allows you to perform document operations in a single request. The body of the request is an array of objects that contains the bulk operations and the target documents to index, create, update, or delete.
+
+### Indexing multiple documents
+
+The following code creates two documents in the `movies` index and one document in the `books` index:
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Index<object>(i => i
+        .Index(movies)
+        .Id(1)
+        .Document(new { Title = "Beauty and the Beast", Year = 1991 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Id(2)
+        .Document(new { Title = "Beauty and the Beast - Live Action", Year = 2017 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Id(1)
+        .Document(new { Title = "The Lion King", Year = 1994 })
+    ));
+```
+
+As you can see, each bulk operation is comprised of two objects. The first object contains the operation type and the target document's `_index` and `_id`. The second object contains the document's data. As a result, the body of the request above contains six objects for three index actions.
+
+Alternatively, the `bulk` method can accept an array of hashes where each hash represents a single operation. The following code is equivalent to the previous example:
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Index<object>(i => i
+        .Index(movies)
+        .Id(1)
+        .Document(new { Title = "Beauty and the Beast", Year = 1991 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Id(2)
+        .Document(new { Title = "Beauty and the Beast - Live Action", Year = 2017 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Id(1)
+        .Document(new { Title = "The Lion King", Year = 1994 })
+    ));
+```
+
+We will use this format for the rest of the examples in this guide.
+
+### Creating multiple documents
+
+Similarly, instead of calling the `create` method for each document, you can use the `bulk` API to create multiple documents in a single request. The following code creates three documents in the `movies` index and one in the `books` index:
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 2", Year = 2030 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 3", Year = 2031 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 4", Year = 2049 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Document(new { Title = "The Lion King 2", Year = 1998 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Document(new { Title = "The Lion King 2", Year = 1998 })
+    ));
```

Comment:
This example is supposed to be using `Create`, and setting movies as the default index, and only specifying books explicitly on the last item. The correct equivalent would be:
```suggestion
var response = await client.BulkAsync(b => b
    .Index(movies)
    .CreateMany(new[]
    {
        new { Title = "Beauty and the Beast 2", Year = 2030 },
        new { Title = "Beauty and the Beast 3", Year = 2031 },
        new { Title = "Beauty and the Beast 4", Year = 2049 }
    })
    .Create<object>(i => i
        .Index(books)
        .Document(new { Title = "The Lion King 2", Year = 1998 })
    ));
```